src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -1796,23 +1796,33 @@ def nsmallest(self, n, columns, keep="first"):  # noqa: PR01, RT01, D200
             )
         )
 
-    @dataframe_not_implemented()
-    def unstack(self, level=-1, fill_value=None):  # noqa: PR01, RT01, D200
+    def unstack(
+        self,
+        level: int | str | list = -1,
+        fill_value: int | str | dict = None,
+        sort: bool = True,
+    ):
         """
         Pivot a level of the (necessarily hierarchical) index labels.
         """
         # TODO: SNOW-1063346: Modin upgrade - modin.pandas.DataFrame functions
-        if not isinstance(self.index, pandas.MultiIndex) or (
-            isinstance(self.index, pandas.MultiIndex)
-            and is_list_like(level)
-            and len(level) == self.index.nlevels
+        # This ensures that non-pandas MultiIndex objects are caught.
+        num_index_levels = self._query_compiler._modin_frame.num_index_levels()
```

Comment:
I'm not sure this is safe, since we don't force a `_modin_frame` on the `_query_compiler`

Reply:
Ah okay let me change this back then, thanks!

Reply:
I recommend using and implementing `self._query_compiler.has_multiindex()`

Reply:
Yep changed to use has_multiindex